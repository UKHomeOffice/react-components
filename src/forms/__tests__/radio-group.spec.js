import React from 'react'
import { mount, render } from 'enzyme'
import RadioGroup from 'govuk-react-components/forms/radio-group';

describe('The Radio Group Component', () => {

  it('should render correctly with no options', () => {
    expect(
      render(<RadioGroup label="test" name="test" options={[]} />)
    ).toMatchSnapshot()
  });

  it('should render correctly with multiple options', () => {
    expect(
      render(<RadioGroup label="test" name="test" options={[{ label: 'opt1-label', value: 'opt1' }, { label: 'opt2-label', value: 'opt2' }]} />)
    ).toMatchSnapshot()
  });

  it('should render correctly with multiple options and one selected', () => {
    expect(
      render(<RadioGroup label="test" name="test" value={'opt2'} options={[{ label: 'opt1-label', value: 'opt1' }, { label: 'opt2-label', value: 'opt2' }]} />)
    ).toMatchSnapshot()
  });

  it('should render correctly as a controlled input with multiple options and one selected', () => {
    expect(
      render(<RadioGroup label="test" name="test" onChange={() => { }} value={'opt2'} options={[{ label: 'opt1-label', value: 'opt1' }, { label: 'opt2-label', value: 'opt2' }]} />)
    ).toMatchSnapshot()
  });

  it('should render a hint when a hint is passed in', () => {
    expect(
      render(<RadioGroup label="test" name="test" options={[]} hint="Testing hint" />)
    ).toMatchSnapshot();
  });

  it('should render an error when an error is passed in', () => {
    expect(
      render(<RadioGroup label="test" name="test" options={[]} error="Testing error" />)
    ).toMatchSnapshot();
  });

})
