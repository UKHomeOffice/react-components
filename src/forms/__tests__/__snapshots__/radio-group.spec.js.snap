// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Radio Group Component should render a hint when a hint is passed in 1`] = `
<div
  class="form-group"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
    <span
      class="form-hint"
    >
      Testing hint
    </span>
  </fieldset>
</div>
`;

exports[`The Radio Group Component should render an error when an error is passed in 1`] = `
<div
  class="form-group form-group-error"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
    <span
      class="error-message"
    >
      Testing error
    </span>
  </fieldset>
</div>
`;

exports[`The Radio Group Component should render correctly as a controlled input with multiple options and one selected 1`] = `
<div
  class="form-group"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
    <div
      class="multiple-choice"
    >
      <input
        id="test-opt1"
        name="test"
        type="radio"
        value="opt1"
      />
      <label
        for="test-opt1"
      >
        opt1-label
      </label>
    </div>
    <div
      class="multiple-choice"
    >
      <input
        checked=""
        id="test-opt2"
        name="test"
        type="radio"
        value="opt2"
      />
      <label
        for="test-opt2"
      >
        opt2-label
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`The Radio Group Component should render correctly with multiple options 1`] = `
<div
  class="form-group"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
    <div
      class="multiple-choice"
    >
      <input
        id="test-opt1"
        name="test"
        type="radio"
        value="opt1"
      />
      <label
        for="test-opt1"
      >
        opt1-label
      </label>
    </div>
    <div
      class="multiple-choice"
    >
      <input
        id="test-opt2"
        name="test"
        type="radio"
        value="opt2"
      />
      <label
        for="test-opt2"
      >
        opt2-label
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`The Radio Group Component should render correctly with multiple options and one selected 1`] = `
<div
  class="form-group"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
    <div
      class="multiple-choice"
    >
      <input
        id="test-opt1"
        name="test"
        type="radio"
        value="opt1"
      />
      <label
        for="test-opt1"
      >
        opt1-label
      </label>
    </div>
    <div
      class="multiple-choice"
    >
      <input
        checked=""
        id="test-opt2"
        name="test"
        type="radio"
        value="opt2"
      />
      <label
        for="test-opt2"
      >
        opt2-label
      </label>
    </div>
  </fieldset>
</div>
`;

exports[`The Radio Group Component should render correctly with no options 1`] = `
<div
  class="form-group"
>
  <fieldset
    class=""
    id="test"
  >
    <legend
      class="form-label-bold"
    >
      test
    </legend>
  </fieldset>
</div>
`;
